infra/ubuntu2204.Dockerfile

```
@@ -0,0 +1,313 @@
+# syntax=docker/dockerfile:1.7-labs
+# This layer represents the bare minimum dependencies needed to run trick sims after they are compiled
+FROM ubuntu:22.04 as minimal-base
+ENV DEBIAN_FRONTEND=noninteractive
+RUN rm -f /etc/apt/apt.conf.d/docker-clean; echo 'Binary::apt::APT::Keep-Downloaded-Packages "true";' > /etc/apt/apt.conf.d/keep-cache
+RUN --mount=type=cache,target=/var/cache/apt,sharing=locked \
+  --mount=type=cache,target=/var/lib/apt,sharing=locked \
+apt update && apt upgrade -y &&  apt install -y \
+    python3-dev \
+    libudunits2-dev \
+    zlib1g-dev \
+    libxml2-dev
+
+FROM minimal-base as minimal-sim-compile-base
+RUN --mount=type=cache,target=/var/cache/apt,sharing=locked \
+  --mount=type=cache,target=/var/lib/apt,sharing=locked \
+apt update && apt upgrade -y &&  apt install -y \
+clang \
+llvm \
+make \
+cmake \
+zip \
+\
+swig \
+g++ \
+libxml2-dev \
+zlib1g-dev \
+llvm-dev \
+libclang-dev \
+libudunits2-dev
+
+FROM minimal-base as koviz-build
+USER 0
+RUN --mount=type=cache,target=/var/cache/apt,sharing=locked \
+  --mount=type=cache,target=/var/lib/apt,sharing=locked \
+  apt install -y vim sudo qtbase5-dev g++ automake make flex bison \
+# note for a production use of this we would want to lock to
+ADD https://github.com/nasa/koviz.git /opt/koviz/
+RUN cd /opt/koviz && qmake && make -j $(nproc) && make install && make clean
```

this is currently an 'open dependency' to HEAD of koviz. it would change if https://github.com/nasa/trick/pull/1772 were merged in:

```suggestion
COPY koviz /opt/koviz
RUN cd /opt/koviz && qmake && make -j $(nproc) && make install && make clean
```

---

1772 does not appear to be a path the trick maintainers would entertain, so we should probably explicitly call out the necessity of the open dependency for future travelers.